### DISCUSSION TITLE
Jest Test Fail - Cannot find module '@web3modal/wagmi/react'

### DISCUSSION THREAD

**Discussion by Evirtual** [WalletConnect/walletconnect-monorepo#4219] [Developer Support] [answered]
Our Jest tests are failing due this error: Cannot find module '@web3modal/wagmi/react'. 

NOTE: We have "@web3modal/wagmi": "^4.0.1" in package.json.

Any Ideas how to solve this?

**Comment by glitch-txs** [comment]
Which package manager are you using?

**Reply by Evirtual** [reply]
We are using yarn

**Reply by glitch-txs** [reply]
Can you try with pnpm?

**Reply by Evirtual** [reply, marked as answer]
Hi @glitch-txs sorry for late reply, I was looking for solution that would not involve to change package manager, for various reasons. But I found a solution that fixed my tests. I added this line in my package.json:
```
"jest": {
 "moduleNameMapper": {
   "^@web3modal/wagmi/react$": "<rootDir>/node_modules/@web3modal/wagmi/dist/esm/exports/react/index.js"
 }
}

**Reply by DanilKasianenko** [reply]
Hello, this solution don't help us I still have error 
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/150665419/8b16f876-03bf-49a1-8521-1f5a7bd50726)

**Reply by glitch-txs** [reply]
it's in the error "projectId is undefined" 😄
